cmd/filebrowser/main.go

```diff
@@ -53,70 +56,82 @@ func init() {
 	flag.StringVarP(&baseurl, "baseurl", "b", "", "Base URL")
-	flag.StringVar(&commands, "commands", "git svn hg", "Default commands option for new users")
 	flag.StringVar(&prefixurl, "prefixurl", "", "Prefix URL")
-	flag.StringVar(&viewMode, "view-mode", "mosaic", "Default view mode for new users")
-	flag.StringVar(&recaptchakey, "recaptcha-key", "", "ReCaptcha site key")
-	flag.StringVar(&recaptchasecret, "recaptcha-secret", "", "ReCaptcha secret")
-	flag.BoolVar(&allowCommands, "allow-commands", true, "Default allow commands option for new users")
-	flag.BoolVar(&allowEdit, "allow-edit", true, "Default allow edit option for new users")
-	flag.BoolVar(&allowPublish, "allow-publish", true, "Default allow publish option for new users")
-	flag.StringVar(&auth.method, "auth.method", "default", "Switch between 'none', 'default' and 'proxy' authentication.")
-	flag.StringVar(&auth.loginHeader, "auth.loginHeader", "X-Forwarded-User", "The header name used for proxy authentication.")
-	flag.BoolVar(&allowNew, "allow-new", true, "Default allow new option for new users")
-	flag.BoolVar(&noAuth, "no-auth", false, "Disables authentication")
-	flag.BoolVar(&alterRecaptcha, "alternative-recaptcha", false, "Use recaptcha.net for serving and handling, useful in China")
-	flag.StringVar(&locale, "locale", "", "Default locale for new users, set it empty to enable auto detect from browser")
 	flag.StringVar(&staticg, "staticgen", "", "Static Generator you want to enable")
 	flag.BoolVarP(&showVer, "version", "v", false, "Show version")
+
+	// User default values
+	flag.StringVar(&commands, "defaults.commands", "git svn hg", "Default commands option for new users")
+	flag.StringVarP(&scope, "defaults.scope", "s", ".", "Default scope option for new users")
+	flag.StringVar(&viewMode, "defaults.viewMode", "mosaic", "Default view mode for new users")
+	flag.BoolVar(&allowCommands, "defaults.allowCommands", true, "Default allow commands option for new users")
+	flag.BoolVar(&allowEdit, "defaults.allowEdit", true, "Default allow edit option for new users")
+	flag.BoolVar(&allowPublish, "defaults.allowPublish", true, "Default allow publish option for new users")
+	flag.BoolVar(&allowNew, "defaults.allowNew", true, "Default allow new option for new users")
+	flag.StringVar(&locale, "defaults.locale", "", "Default locale for new users, set it empty to enable auto detect from browser")
+
+	// Recaptcha settings
+	flag.StringVar(&recaptcha.host, "recaptcha.host", "https://www.google.com", "Use another host for ReCAPTCHA. recaptcha.net might be useful in China")
+	flag.StringVar(&recaptcha.key, "recaptcha.key", "", "ReCaptcha site key")
+	flag.StringVar(&recaptcha.secret, "recaptcha.secret", "", "ReCaptcha secret")
+
+	// Auth settings
+	flag.StringVar(&auth.method, "auth.method", "default", "Switch between 'none', 'default' and 'proxy' authentication")
+	flag.StringVar(&auth.header, "auth.header", "X-Forwarded-User", "The header name used for proxy authentication")
```

Comment:
Can we make some deprecated flags? For example when user tries to use the old `--allow-new` flag, we give a hint that `--allow-new is deprecated, use --defaults.allowNew instead`.

Reply:
I think we could, but we'd need to do it for every flag. If you're at it, you can update this PR itself 😄

Reply:
I am at it. Will push soon.

https://godoc.org/github.com/spf13/pflag#FlagSet.MarkDeprecated